test/extensions/strings_test.dart

```diff
@@ -35,5 +35,16 @@ void main() {
   test('extension alphanumeric negative numbers', () {
     final test = '-4';
     expect(test.alphanumeric, 'Negative4');
+    final test2 = '-4-grey';
+    expect(test2.alphanumeric, 'Negative4Grey');
+    final test3 = '--4';
+    expect(test3.alphanumeric, '4');
```

Comment:
Thank you for including this test. I haven't seen cases like these used, so I'm not sure what the intention would be. This keeps it the same as the pre-existing code so it's the safest option imo. 👍